dartagnan/src/main/java/com/dat3m/dartagnan/program/event/pthread/Create.java

```diff
@@ -53,6 +55,8 @@ public Create getCopy(){
 
     @Override
     protected RecursiveFunction<Integer> compileRecursive(Arch target, int nextId, Event predecessor, int depth) {
+    	Preconditions.checkArgument(Arrays.asList(Arch.values()).contains(target), "Compilation to " + target + " is not supported for " + this);
```

Comment:
Again, this is just a complex null check. Use `Preconditions.checkNotNull` or `Precondtions.checkArgument(target != null)`.
I think I would prefer the latter in this case.